### DISCUSSION TITLE
Command's choice() method returning keys instead of values from options array.

### DISCUSSION THREAD

**Discussion by CaalGooR** [laravel/framework#51391] [Q&A]
### Laravel Version

11

### PHP Version

8.3

### Description

I'm experiencing an unusual behavior with the choice() method of the Command class in Laravel. As per Laravel's documentation, when using choice() with an associative array, it should return the value for a selected option. However, in my case, it's returning the key from the array.

Here's a code sample illustrating this:

```php
private function requestRegimenFiscal(): RegimenFiscal
{
    $regimenFiscal = '';

    // Regiment fiscal list
    $regimenFiscalList = [];

    foreach (RegimenFiscal::cases() as $case) {
        $regimenFiscalList[$case->label()] = $case->value;
    }

    while (! (RegimenFiscal::tryFrom($regimenFiscal) instanceof RegimenFiscal)) {
        $regimenFiscal = $this->choice(
            'Select the regimen fiscal number of the receptor',
            $regimenFiscalList
        );

        dump($regimenFiscal);
    }

    return RegimenFiscal::from($regimenFiscal);
}
```

In this case, $case->label() is a long text and $case->value is a number so the $regimenFiscalList array is built with long text as the key and number as the value.
When an option is selected via the $this->choice() call, the returned value is the long text (the array's key), not the number (the value).
Any assistance looking into this issue would be much appreciated.

![image](https://github.com/laravel/framework/assets/32117443/fa1b7f39-de7c-4f14-a441-f9c8c27b058d)

### Steps To Reproduce
```php

$value = $this->choice('Bug Example', [
            "Here's my long text 1" => 1,
            "Here's my long text 2" => 2,
            "Here's my long text 3" => 3,
            "Here's my long text 4" => 4,
        ]);


```
![image](https://github.com/laravel/framework/assets/32117443/dd859250-52ac-46b7-a0f1-b925714f9bc2)

**Comment by bturski** [comment]
I use `$this->output->askQuestion()` in commands instead of `choice()` if I need to enter and return the id. 

Source: https://stackoverflow.com/questions/64656614/getting-an-id-from-laravel-command-choice
> ```PHP
> use Symfony\Component\Console\Question\ChoiceQuestion;
> 
> $question = 'Select the category:';
> 
> $choices = [
>     1 => 'Category 1',
>     2 => 'Category 2',
>     3 => 'Category 3',
>     4 => 'Category 4',
> ];
> 
> $question = new ChoiceQuestion($question, $choices);
> $question->setValidator(function($res)
> {
>     // here is where laravel interpolates things, but we will only return the selected choice id.
>     return $res;
> });
> 
> $id = $this->output->askQuestion($question);
> 
> $text = $choices[$id];

Hope it helps

**Comment by monkeyphysics** [comment]
`$this->choice()` is an awkward function in how it returns the value, unless the keys are strings, in which case it returns the key.

I've found that using [Laravel Prompts](https://laravel.com/docs/12.x/prompts#select) `select()` function is much better.

(The Laravel Prompts package is already installed with modern Laravel versions anyway...)
